.github/workflows/ci.yml

```diff
@@ -10,17 +10,46 @@ on:
 jobs:
   build:
     runs-on: ubuntu-latest
+    continue-on-error: ${{ matrix.experimental }}
+    strategy:
+      fail-fast: false
+      matrix:
+        ghc:
+          - "ghc884"
+          - "ghc8107"
+          - "ghc901"
+          - "ghc921"
+        experimental: [false]
+        include:
+          - ghc: "ghcHEAD"
+            experimental: true
+    steps:
+    - name: Install nix
+      uses: cachix/install-nix-action@v16
+      with:
+        nix_path: nixpkgs=channel:nixos-unstable
+    - name: Checkout repositiory code
+      uses: actions/checkout@v3
+    - name: List files
+      run: ls ${{ github.workspace }}
+    - name: Cat flake.nix
+      run: cat flake.nix
+    - name: dry-run
+      run: nix build --dry-run --print-build-logs .#${{ matrix.ghc }}_all || ${{ matrix.experimental }}
+    - name: build
+      run: nix build --print-build-logs -j1 .#${{ matrix.ghc }}_all || ${{ matrix.experimental }}
+  # We currently want to maintain at least non-nix build with up-to-date hackage index to check
+  # the traditional builds keeps working.
+  non-nix-build:
+    runs-on: ubuntu-latest
+    continue-on-error: true
```

Comment:
If we’re keeping `|| true`, should we remove this?

Reply:
removed